frontend/server/src/main/java/com/amazonaws/ml/mms/util/ConfigManager.java

```diff
@@ -122,6 +122,10 @@ public ConfigManager(Arguments args) {
         if (models != null) {
             prop.setProperty(LOAD_MODELS, String.join(",", models));
         }
+
+        if (!prop.containsKey(NUMBER_OF_GPU)) {
```

Comment:
We shouldn't use GPU if the properties doesn't contain GPU key. Customers might not want to use GPUs.

Reply:
There should be a special value (-1) or some such, which should mean auto-detect GPUs.

Reply:
1. number_of_gpu = 0 is still a valid configuration.
2. If user didn't config gpu, auto detect should be the default. What's the use case we don't want auto detect as default?
3. We don't need special value (-1), user didn't add this properties in config file is the indicator of auto detection.

Reply:
1. It should be valid.
2. Auto-detection shouldn't happen if customers don't want to use num GPUs option. We don't understand the customer use case. They might be running training on the same host and GPUs might be used for that and they might want to run inference only on CPUs. So, we should make it explicit to use GPUs and not auto-detect. 
3. If we go by point 2, then we would need a value for auto-detect. The value could be whatever we choose.
4. We should do auto-detect for any value larger than 0. The actual value should be [0,  min(given_number_of_gpu, total_available_gpu)] if a value is given.